### PR TITLE
feat: [CDS-78339]:k8s-apply-remtoe-manifest

### PR DESCRIPTION
**Pull request by litvinovstas**
### Summary
1. Added option for K8sApply step, to switch between local and  remote manifest
2. Added option to provide remote K8s manifest.

Test cases:
1. Added manifest from remote repo (with fixed value).
2. Added K8s manifest with runtime values.
3. Save and runned pipeline

Screenshot 2023-11-14 at 23 44 09
Screenshot 2023-11-14 at 23 44 42

#### Screenshots
Screenshot 2023-11-14 at 23 43 38

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
- MessageMetadata: `trigger messagecheck`
- GitLeaks: `trigger gitLeaks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

### CODE REVIEW

File: src/modules/70-pipeline/components/ManifestSelection/ManifestInterface.ts [hunk 1]
```diff
@@ -106,11 +107,21 @@ export interface ManifestListViewProps {
...
 
+export type K8sManifest = 'K8sManifest'
+
+export interface ManifestSourceK8s extends Omit<ManifestSourceWrapper, 'type'> {
+  type: string
+}
+
 export interface ManifestStepInitData {
-  connectorRef: string | undefined | ConnectorSelectedValue
+  connectorRef?: string | undefined | ConnectorSelectedValue
```
Comment (Sagar-Rajput7) [inline src/modules/70-pipeline/components/ManifestSelection/ManifestInterface.ts:117]:
```suggestion
  connectorRef?: string | ConnectorSelectedValue
```

File: src/modules/75-cd/components/PipelineSteps/K8sApply/K8sApplyStep.tsx [hunk 2]
```diff
@@ -590,7 +718,7 @@ export class K8sApplyStep extends PipelineStep<K8sApplyData> {
     name: '',
     type: StepType.K8sApply,
     spec: {
-      filePaths: [],
+      // filePaths: [],
```
Comment (Sagar-Rajput7) [inline src/modules/75-cd/components/PipelineSteps/K8sApply/K8sApplyStep.tsx:721]: nit: remove comment if not needed

File: src/modules/75-cd/components/PipelineSteps/K8sApply/K8sRemoteManifest/__tests__/K8sRemote.test.tsx [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+  })
+
+  test('initial rendering runtime values', async () => {
+    const { container } = render(
+      <TestWrapper>
+        <K8sRemoteFile {...props} prevStepData={initialRuntimeValues} />
+      </TestWrapper>
+    )
+
+    const modals = document.getElementsByClassName('bp3-dialog')
```
Comment (Sagar-Rajput7) [inline src/modules/75-cd/components/PipelineSteps/K8sApply/K8sRemoteManifest/__tests__/K8sRemote.test.tsx:157]: can use - `findDialogContainer()`

File: src/modules/75-cd/components/PipelineSteps/K8sApply/K8sRemoteManifest/__tests__/K8sRemote.test.tsx [hunk 4]
```diff
@@ -0,0 +1,219 @@
...
+              }
+            },
+            store: 'Git',
+            selectedManifest: ManifestDataType.K8sManifest
+          }}
+        />
+      </TestWrapper>
+    )
+
+    const modals = document.getElementsByClassName('bp3-dialog')
```
Comment (Sagar-Rajput7) [inline src/modules/75-cd/components/PipelineSteps/K8sApply/K8sRemoteManifest/__tests__/K8sRemote.test.tsx:203]: same

File: src/modules/75-cd/components/PipelineSteps/K8sApply/K8sRemoteManifest/__tests__/K8sRemoteManifest.test.tsx [hunk 5]
```diff
@@ -0,0 +1,176 @@
...
+  test('initial rendering', async () => {
+    const { container } = render(
+      <TestWrapper>
+        <TestComponent initialValues={{ spec: { ...initialValues } }} store="Git" isReadonly={false} />
+      </TestWrapper>
+    )
+    const addFileButton = await findByText(container, 'cd.manifestSelectPlaceHolder')
+    expect(addFileButton).toBeDefined()
+    await userEvent.click(addFileButton)
+    const portal = document.getElementsByClassName('bp3-dialog')[0]
```
Comment (Sagar-Rajput7) [inline src/modules/75-cd/components/PipelineSteps/K8sApply/K8sRemoteManifest/__tests__/K8sRemoteManifest.test.tsx:92]: same here

File: src/modules/75-cd/components/PipelineSteps/K8sApply/K8sRemoteManifest/__tests__/K8sRemoteManifest.test.tsx [hunk 6]
```diff
@@ -0,0 +1,176 @@
+import React from 'react'
```
Comment (Sagar-Rajput7) [inline src/modules/75-cd/components/PipelineSteps/K8sApply/K8sRemoteManifest/__tests__/K8sRemoteManifest.test.tsx:1]: NIT: add polyform license to the newly added files

**Review by Sagar-Rajput7 — APPROVED** [review]

**Review by satyam-harness — APPROVED** [review]

**Review by Sagar-Rajput7 — DISMISSED** [review]

**Review by satyam-harness — APPROVED** [review]

**Review by Sagar-Rajput7 — APPROVED** [review]

**Pull request merged**